### PR TITLE
Feanil/unpin drf

### PR DESCRIPTION
**Pull request by feanil**

### CODE REVIEW

**Comment by davestgermain** [comment]
Dupe of #564?

**Comment by feanil** [comment]
@davestgermain ah, I didn't see that Ned had already started.  I'll probably close that one and keep this one as it has other modernization changes on it.  Thanks for the heads up.

File: requirements/constraints.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Version constraints for pip installation.
```
Comment (nasthagiri) [inline requirements/constraints.txt]: Nice!  Is a `constraints.txt` file something we've used before as well?

File: requirements/base.in [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # Core requirements for using this application
+-c constraints.txt
 
 # Django/Framework Packages
 django<2,>=1.11
 django-model-utils>=2.3.1
-djangorestframework>=3.1,<3.7
+djangorestframework
```
Comment (nasthagiri) [inline requirements/base.in]: Should this be pinned by a major version number?

File: requirements/constraints.txt (hunk 1 above)
Comment (feanil) [reply requirements/constraints.txt]: Yea,  we started using it in edx-platform a little while ago when Jeremy first discovered it.  But we're slowly pushing it is slowly spreading.  At this point, we should probably make it a part of our best practice and update the OEP.  I'll put in a ticket on the BOM backlog to do that.

File: requirements/base.in (hunk 2 above)
Comment (feanil) [reply requirements/base.in]: The tests currently test all current versions and will automatically test the latest version so I feel pretty comfortable not to pin it here.  If the next major version breaks this app, the tests will pick it up on the automated `make upgrade` PR once the new version is available.  Pinning to the major version here would mean that we would need to come back and unpin here before we could upgrade edx-platform, and we would need to do the same in all apps that depend on DRF.  This felt too cumbersome to me.  Let me know if you disagree and we should change this strategy.

**Comment by feanil** [comment]
Merging now, and we can always come back and update if we wont to pin to the major version.

**Pull request merged**